Controller/Adminhtml/Credentials/Check.php

```diff
@@ -81,18 +111,26 @@ public function execute(): Json
     private function testCredentials(): ?ClientInterface
     {
         $config = $this->getCredentials();
+        $mode = $this->getRequest()->getParam('mode');
 
         if (!$config['credentials']['client_id']) {
-            throw new LocalizedException(__('No Client ID set!'));
+            throw new LocalizedException(__('Client Id is missing'));
         }
 
         if (!$config['credentials']['client_secret']) {
-            throw new LocalizedException(__('No Client Secret set!'));
+            throw new LocalizedException(__('Client Secret is missing'));
+        }
+
+        if (!$config['credentials']['private_key'] || $this->getRequest()->getParam('delete_private_key') === 'true') {
```

Comment:
Perhaps I'm confusing myself here, but wouldn't this always through if the delete private key checkbox is ticked?

Reply:
If the delete private key checkbox is ticked, this will always through, that is correct. Saving the file happens before deleting, so having a simultaneous file upload and this checkbox results in having the key file deleted at the end, if you were to click "save" on the form in this state.